src/ILCompiler.Compiler/src/IL/ILImporter.Scanner.cs

```diff
@@ -993,6 +1014,21 @@ private int ReadILTokenAt(int ilOffset)
                 + (_ilBytes[ilOffset + 3] << 24));
         }
 
+        private void ReportOutOfRangeBranchTarget(int targetOffset)
+        {
+            throw new TypeSystemException.InvalidProgramException();
+        }
+
+        private void ReportFallthrough()
```

Comment:
It is a bit confusing that `fallthrough` is used for both the valid condition (fallthrough from one basic block to another one) and for the error condition (fallthrough at the end of the method). Maybe the error condition can use longer or different name.